proposals/nnnn-structured-concurrency.md

```diff
@@ -570,6 +572,10 @@ let dinnerHandle: Task.Handle<Meal, Error> = Task.runDetached {
 try await eat(dinnerHandle)
 ```
 
+By default, the new task will be initially scheduled on the default global
+concurrent executor. `runDetached` may optionally be given a `startingOn`
+executor on which to schedule the new task instead.
```

Comment:
That sounds good; 

I wonder if we should also have `runDetached(on: (self where self is an Actor))`? That could provide static guarantees that it it safe to access any actor isolated state from this detached task 🤔 

// cc @DougGregor

Reply:
AIUI actor-isolated operations should cause the task to automatically hop executors to somewhere it's safe to schedule work for the actor.

Reply:
github ate my example when you updated the PR :) 

The question I have in mind is very specific, about state not actor functions which I know cause hops:

```
runDetached / group.add { // (hop 1) some executor
await actor.hi() // hop to actor (hop 2)
// (hop 3) return to some executor to return
}
```

Note that in such simple closures which _only_ call an actor func, hops 1 and 3 are very unnecessary. The question in this example is -- do we guarantee that hop 1 happens? Hop 3 is very useless and we can definitely optimize it away without much semantic impact.

Hop 1 we have to explicitly decide what to do about, because the group's rule about "executor may suspend the add". What I'd suggest is:

- if no `startingOn` is set, we're allowed to not even perform hop 1
  - by default it goes to global shared executor; can we assume it would never backpressure an `add`? That seems likely...
  - thus... we could avoid the hop 1 (if we are guaranteed global executor never would want to suspend an add)
- if executor is set explicitly, we always hop to it (hop 1)
  - we should be allowed not to do hop 3 though

Sounds reasonable?

Now, the actual question is about properties i.e. "just state":

```
runDetached { 
  actor.varProperty = 20
}
```

so... there's a few answers here:

- can we we spell "run this on that actor please" or do we not want to allow such spelling?
  - it could be `run...(on: self) { let x = await thingy(); /*hops back to self executor*/ self.x = x }` this would work if we allow this
  - or are we saying a closure is allowed to be marked on a specific actor, like `run... { (actor self) in ... }` which causes the right hops?
  - OR are we saying this is a terrible pattern and please don't do this -- make functions to do such setting or `var` reading?

Either can be fine but we didn't address it much so far and I guess it relates to how we want to spell those "on:" things